### PR TITLE
Upgrading etcd client/api to v3

### PR DESCRIPTION
**Pull request by ankitrgadiya**
The watcher is currently using Etcd client v3.3.25. That's an incompatible Go module which the upstream fixed in the minor v3.5.0. This is causing issues while using the Watcher in a Go modules enabled repository.

This PR upgrades the etcd to same major version's Go modules compatible version to fix the issue.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,17 +4,6 @@ go 1.16
...
-	github.com/envoyproxy/go-control-plane v0.9.5 // indirect
-	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/golang/protobuf v1.5.2 // indirect
-	github.com/google/uuid v1.2.0 // indirect
-	github.com/stretchr/testify v1.5.1 // indirect
-	go.etcd.io/etcd v3.3.25+incompatible
-	go.uber.org/zap v1.16.0 // indirect
-	google.golang.org/grpc v1.26.0 // indirect; updating this breaks etcd
+	go.etcd.io/etcd/api/v3 v3.5.0
+	go.etcd.io/etcd/client/v3 v3.5.0
```
Comment (hsluoyz) [inline go.mod:8]: Why there's no indirect dependency?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (ankitrgadiya) [reply go.mod:8]: The previous version v3.3.25 wasn't a Go module so its dependencies were listed as indirect. That's why +incompatible as well.
Now v3.5.0 is a Go module and it lists its dependencies properly in its own go.mod. So these are no longer indirect dependencies. Instead these are listed in go.sum.

https://golangbyexample.com/direct-indirect-dependency-module-go/

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@closetool plz review.

**Review by closetool — COMMENTED** [review]
Does this change has forward compatibility?

**Comment by hsluoyz** [comment]
@closetool good idea. How to trigger a major release (v2.0.0) via semantic-release? What prefix?

**Comment by closetool** [comment]
@hsluoyz @ankitrgadiya 
https://github.com/semantic-release/semantic-release#commit-message-format
`BREAKING CHANGE` or `perf`

**Comment by hsluoyz** [comment]
@ankitrgadiya plz change it.

**Comment by ankitrgadiya** [comment]
> @hsluoyz @ankitrgadiya
> https://github.com/semantic-release/semantic-release#commit-message-format
> `BREAKING CHANGE` or `perf`

Hey, I'm not sure if `BREAKING CHANGE` should be in the header of the commit or in the Body. `perf` doesn't seem appropriate here. Conventional Commits says `BREAKING CHANGE` needs to be in the Body.

**Comment by closetool** [comment]
@ankitrgadiya It says `Commit message format`

**Comment by ankitrgadiya** [comment]
@closetool The commit message has two parts header and body. I tried adding it in the header or first line but that seems to be making the CI upset. I checked the analyser semantic-release is using and it will parse BREAKING CHANGE from the body as well so that's what I did. This follows the convernsion in Conventional Commits as well which CI is using.

**Comment by closetool** [comment]
@ankitrgadiya Ok, thanks for your trying.

**Review by closetool — APPROVED** [review]
LGTM

**Comment by hsluoyz** [comment]
@ankitrgadiya still "fix:"?

**Comment by ankitrgadiya** [comment]
@hsluoyz We're supposed to add the BREAKING CHANGE in the commit message. If you expand the commit message then you'll see it there :)

**Comment by ankitrgadiya** [comment]
@hsluoyz @closetool One thought, should we make the Go module as /v2 as well since we are planning to bump the major version?

**Comment by hsluoyz** [comment]
@ankitrgadiya sounds good.

About commit message, why not do:

image

**Comment by ankitrgadiya** [comment]
@hsluoyz I tried that first but the Semantic Pull Request check was failing because of that.

**Comment by hsluoyz** [comment]
@ankitrgadiya OK, then do this:

> One thought, should we make the Go module as /v2 as well since we are planning to bump the major version?

**Comment by closetool** [comment]
@hsluoyz That will make semantic-pull-requests failed
https://github.com/zeke/semantic-pull-requests#configuration

**Comment by ankitrgadiya** [comment]
@hsluoyz @closetool Can you check now? I also updated the README with a working example

**Review by closetool — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ func updateCallback(rev string) {
 func main() {
     // Initialize the watcher.
     // Use the endpoint of etcd cluster as parameter.
-    w, _ := etcdwatcher.NewWatcher("http://127.0.0.1:2379")
+    w, _ := etcdwatcher.NewWatcher([]string{"http://127.0.0.1:2379"}, "keyname")
```
Comment (hsluoyz) [inline README.md:29]: What's `"keyname"`?

Comment (ankitrgadiya) [reply README.md:29]: A placeholder for the key to watch on Etcd

**Comment by hsluoyz** [comment]
Thanks for the contribution!

**Pull request merged**